bindings/azure/storagequeues/storagequeues_test.go

```diff
@@ -0,0 +1,175 @@
+	"github.com/stretchr/testify/mock"
+)
+
+type MockHelper struct {
+	mock.Mock
+}
+
+func (m *MockHelper) Init(accountName string, accountKey string, queueName string) error {
+	retvals := m.Called(accountName, accountKey, queueName)
+	return retvals.Error(0)
+}
+
+func (m *MockHelper) Write(data []byte) error {
+	return nil
+}
+
+func (m *MockHelper) Read(ctx context.Context, consumer *consumer) error {
+	return nil
+}
+
+func TestWriteQueue(t *testing.T) {
+
+	mm := new(MockHelper)
+	mm.On("Init", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(nil)
+
+	a := AzureStorageQueues{helper: mm}
+
+	m := bindings.Metadata{}
+	m.Properties = map[string]string{"accountKey": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==", "queueName": "queue1", "accountName": "devstoreaccount1"}
+
+	err := a.Init(m)
+	assert.Nil(t, err)
+
+	r := bindings.WriteRequest{Data: []byte("This is my message")}
+
+	err = a.Write(&r)
+
+	assert.Nil(t, err)
+}
+
+// Uncomment this function to write a message to local storage queue
+/* func TestWriteLocalQueue(t *testing.T) {
+
+	a := AzureStorageQueues{helper: &AzureQueueHelper{reqURI: "http://127.0.0.1:10001/%s/%s"}}
+
+	m := bindings.Metadata{}
+	m.Properties = map[string]string{"accountKey": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==", "queueName": "queue1", "accountName": "devstoreaccount1"}
+
+	err := a.Init(m)
+	assert.Nil(t, err)
+
+	r := bindings.WriteRequest{Data: []byte("This is my message")}
+
+	err = a.Write(&r)
+
+	assert.Nil(t, err)
+} */
+
+func TestReadQueue(t *testing.T) {
+	mm := new(MockHelper)
+	mm.On("Init", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(nil)
+
+	a := AzureStorageQueues{helper: mm}
+
+	m := bindings.Metadata{}
+	m.Properties = map[string]string{"accountKey": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==", "queueName": "queue1", "accountName": "devstoreaccount1"}
+
+	err := a.Init(m)
+	assert.Nil(t, err)
+
+	r := bindings.WriteRequest{Data: []byte("This is my message")}
+
+	err = a.Write(&r)
+
+	assert.Nil(t, err)
+
+	var handler = func(data *bindings.ReadResponse) error {
+		s := string(data.Data)
+		assert.Equal(t, s, "This is my message")
+		return nil
+	}
+
+	go a.Read(handler)
+
+	time.Sleep(5 * time.Second)
+
+	pid := syscall.Getpid()
+	proc, _ := os.FindProcess(pid)
+	proc.Signal(os.Interrupt)
+}
+
+// Uncomment this function to test reding from local queue
+/* func TestReadLocalQueue(t *testing.T) {
+	a := AzureStorageQueues{helper: &AzureQueueHelper{reqURI: "http://127.0.0.1:10001/%s/%s"}}
+
+	m := bindings.Metadata{}
+	m.Properties = map[string]string{"accountKey": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==", "queueName": "queue1", "accountName": "devstoreaccount1"}
```

Comment:
Is this secret meant to be here..?

Reply:
Its the storage emulator fixed secret. Its harmless...